app/src/main/java/com/kenchen/capstonenewsapp/views/news/NewsViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package com.kenchen.capstonenewsapp.views.news
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.kenchen.capstonenewsapp.App.Companion.remoteApi
+import com.kenchen.capstonenewsapp.model.Article
+import com.kenchen.capstonenewsapp.networking.RemoteResult
+import com.kenchen.capstonenewsapp.utils.mapException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class NewsViewModel : ViewModel() {
+
+    private val _headLineNewsLiveData = MutableLiveData<RemoteResult<List<Article>>>()
+    val headLineNewsLiveData: LiveData<RemoteResult<List<Article>>> = _headLineNewsLiveData
+
+    private val test = MutableLiveData<List<String>>()
+
+    private val _newsLoadingStateLiveData = MutableLiveData<NewsLoadingState>()
+    val newsLoadingStateLiveData: LiveData<NewsLoadingState> = _newsLoadingStateLiveData
```

Comment:
Great work here including a state livedata! One adjustment you could do is make the `NewsLoadingState` a `sealed class` as well.  This will allow you to remove the need for passing back the `RemoteResult` back to the activity.  In other words, the `RemoteResult` is useful for returning data from the networking layer to the viewmodel, and the viewmodel can convert the data to a "state" object that the view can use:

```kotlin
sealed class NewsState {
    object Loading : NewsState()
    data class Loaded(articles: List<Article>) : NewsState()
    data class Error(errorMessage: String) : NewsState()
}
```

This way, you will have just one livedata object exposed back to the view, and all the required data is there in one place.

This is merely a suggestion though!

Reply:
I like your suggestion. This makes the code cleaner and follows the separation of concern.